pkgs/development/python-modules/flask-migrate/default.nix

```diff
@@ -1,25 +1,47 @@
-{ lib, buildPythonPackage, fetchPypi, isPy3k, glibcLocales, flask, flask_sqlalchemy, flask_script, alembic }:
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, isPy27
+, alembic
+, flask
+, flask_script
+, flask_sqlalchemy
+, python
+}:
 
 buildPythonPackage rec {
   pname = "Flask-Migrate";
   version = "3.1.0";
+  format = "setuptools";
+  disabled = isPy27;
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "57d6060839e3a7f150eaab6fe4e726d9e3e7cffe2150fb223d73f92421c6d1d9";
+  src = fetchFromGitHub {
+    owner = "miguelgrinberg";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0zj7qpknvlhrh4fsp5sx4fwyx3sp41ynclka992zympm3xym9zyq";
   };
 
-  checkInputs = [ flask_script ] ++ lib.optional isPy3k glibcLocales;
-  propagatedBuildInputs = [ flask flask_sqlalchemy alembic ];
+  propagatedBuildInputs = [
+    alembic
+    flask
+    flask_sqlalchemy
+  ];
 
-  # tests invoke the flask cli which uses click and therefore has py3k encoding troubles
-  preCheck = lib.optionalString isPy3k ''
-    export LANG="en_US.UTF-8"
+  checkInputs = [
+    flask_script
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+    ${python.interpreter} -m unittest discover
```

Comment:
Is there a reason we can't use pytestCheckHook here?

Reply:
Yes, it uses pythons unittest module, not pytest.

Reply:
Many packages can still be run through pytest even if it's not their official supported test runner. If pytest still doesn't work is there a reason that setuptoolsCheckHook doesn't work?

Reply:
The `setuptoolsCheckHook` uses `python setup.py test` which is deprecated.

> WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point indep

We are mimicking the upstreams test environment. https://github.com/miguelgrinberg/Flask-Migrate/blob/943a938384deafcd1dedb10ead107827cf3324c9/tox.ini#L16